sample/src/main/java/no/finn/rxactivityresponse/sample/SimplePermissionButton.java

```diff
@@ -0,0 +1,55 @@
+package no.finn.rxactivityresponse.sample;
+
+import java.lang.ref.WeakReference;
+
+import android.Manifest;
+import android.app.Activity;
+import android.content.Context;
+import android.util.AttributeSet;
+import android.view.View;
+import android.widget.Button;
+import android.widget.Toast;
+
+import no.finn.android.rx.RxPermission;
+import no.finn.android.rx.RxPermissionRationale;
+import no.finn.android.rx.RxState;
+import no.finn.android.rx.RxStateRestart;
+
+import junit.framework.Assert;
+import rx.functions.Action1;
+
+public class SimplePermissionButton extends Button implements View.OnClickListener, RxStateRestart {
+    private final RxState rxState;
+
+    public SimplePermissionButton(Context context, AttributeSet attrs) {
+        super(context, attrs);
+        // The State has to be created construction time (it's the only way we can get if we're serialized mid flow!)
+        rxState = RxState.get(context, ActivityResponses.GET_LOCATIONPERMISSION, new WeakReference<RxStateRestart>(this));
```

Comment:
The WeakReference is the part I'm the least happy with.

The problem is that we need to hold a referance to the view (to return a response), but if the view is destroyed without the user clicking anything we never get a callback.

Option 1 : We could tell every implementation to explicitly create RxState in onAttachedToWindow, and destroy it in onDetachedFromWindow, not a huge fan... but it might be cleaner than this :/. If used in an activity, where would you delink? Has to be before onResume...

Option 2: Keep the WeakReference, it makes it fairly explicit that if you use a inner class here things will crash. (I also throw an exception with that specific case as an explanation if this disapears before we expect it to).

Option 3: Create some more get calls that allows sending in a View/Activity/Fragment. This probably covers the 90% case, and those should not disapear before we expect them to. As they are being used by the system up until the point where we need them.